src/integrationTest/java/uk/gov/hmcts/reform/sendletter/services/LetterServiceTest.java

```diff
@@ -50,6 +54,42 @@ public void generates_and_saves_pdf() throws IOException {
         document.validate();
     }
 
+    @Test
+    public void generate_and_save_pdf_and_return_same_id_on_resubmit() throws IOException {
+        // given
+        LetterRequest sampleRequest = SampleData.letter();
+        UUID id1 = service.send(sampleRequest, SERVICE_NAME);
+        Letter result = letterRepository.findOne(id1);
+
+        // and
+        assertThat(result.getState()).isEqualByComparingTo(LetterState.Created);
+
+        // then
+        UUID id2 = service.send(sampleRequest, SERVICE_NAME);
+
+        // and
+        assertThat(id1).isEqualByComparingTo(id2);
```

Comment:
also check that only one row was insterted (with currently implementation you save a new one anyway)

Reply:
is table identification comparison enough?